server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -104,14 +125,30 @@ public FlatObjectField(String field, BytesRef term, FieldType ft) {
      * @opensearch.internal
      */
     public static class Builder extends FieldMapper.Builder<Builder> {
-
-        public Builder(String name) {
+        private NamedAnalyzer normalizer;
+        private int ignoreAbove = Integer.MAX_VALUE;
+        private String nullValue;
+        private SimilarityProvider similarity;
+        private int depthLimit = Integer.MAX_VALUE;
+        private final IndexAnalyzers indexAnalyzers;
+
+        public Builder(String name, IndexAnalyzers indexAnalyzers) {
             super(name, Defaults.FIELD_TYPE);
             builder = this;
+            this.indexAnalyzers = indexAnalyzers;
+        }
+
+        public void setNormalizer(String normalizer) {
+            this.normalizer = this.indexAnalyzers.getNormalizer(normalizer);
+            this.indexAnalyzer = this.normalizer;
```

Comment:
When normalizer is set, it will be used in index and search, example in keyword type: https://github.com/opensearch-project/OpenSearch/blob/c71fd4a2f2e6d5d7d9f2f304c573180027af8f44/server/src/main/java/org/opensearch/index/mapper/KeywordFieldMapper.java#L281

Did I understand wrong?